src/library/Navigation/types.js

```diff
@@ -0,0 +1,215 @@
+/* @flow */
+import { ALIGN, INTERNAL_TYPE, PREFIX, TYPE } from './constants';
+
+import type {
+  ComponentTheme,
+  ComponentThemeFn,
+  ThemeValue
+} from '../themes/types';
+import type { MenuItemRenderFn, MenuItemType } from '../Menu/types';
+
+// [1] These purposefully reference $Values instead of $Keys
```

Comment:
I am not sure that `$Values` is doing what you expect here.

For example, try adding the following snippet in this file (anywhere after line 20).

```js
// Expectedly causes flow error because "foo" is not a key in `ALIGN`
const foo: Align = 'foo';
console.log(foo);

// Does not cause flow error because `Prefix` type is actually just `string` and not a union of `PREFIX` values.  I am not sure how to get the equivalent values in Flow;
const bar: Prefix = 'bar';
console.log(bar);
```

Reply:
Hmm... it works if we `type` (instead of `const`) the mapping object:

https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgAqAogMqFgC8YA3qmGHAHZ4BcYA5APrN4cA09MBgRx23EXA6oAvgG502fGABqAQxgBXPAGcqYACTqtugDwlyAPgWpgwMADk4YPACdXcV6gDGzHRjAANw1tAEZ2Y209am5eDhs7MGJ3Tx8-AOCTACYIkN19DjgMAAs3eNQgA

Thoughts?

Reply:
Oh wait, I think I found a better way.  Try this...

https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjOA7AzgFzADEB5EsAXjBICMArAUw3wDooAnBhgLwYAoA3qjBgaAQ3YAuMAHJx7GQBphosbmly1M1AF8AlAG50+AJ4AHBsThxKYACQA1MTACuDXAB5TFuFGJkAPhVUYGAwAHU4dgBrXABCTBwCMAA3AEZpImtbTQUjELCAUXZ2KNwwNTAGAA8LZgYAE0S8QhSAJkzsqhkARxcXapkjIA

Also probably relevant to your curiosity - https://kangax.github.io/compat-table/es5/#test-Object.freeze